functionary/ui/views/environment_select.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.exceptions import PermissionDenied
+from django.http import HttpRequest
+from django.shortcuts import get_object_or_404, render
+from django.views.generic import TemplateView
+from django_htmx import http
+
+from core.models import Environment
+from ui.views.utils import set_session_environment
+
+
+class EnvironmentSelectView(LoginRequiredMixin, TemplateView):
+    def post(self, request: HttpRequest):
+        pk = request.POST["environment_id"]
+        environment = get_object_or_404(Environment, pk=pk)
```

Comment:
I changed this just for you.  Since the resultant behavior of your suggestion would be that a 404 is never returned, I just restructured it to not use `get_object_or_404` at all.